### ISSUE TITLE
Is it possible to apply 3DGS with depth regularition?

### ISSUE THREAD

**Issue by Vilour** [Linketic/CityGaussian#137]
Thank you for your great work! If I want to get 3DGS model with the regularization of Depth-Anything, how can I modify the config file?

**Comment by MetasoftPro** [comment]
> with the regularization of Depth-Anything

[Depth Anything](https://depth-anything.github.io/) is not good enough compared to [Depth Anything V2](https://depth-anything-v2.github.io/) nowadays

_Depth Regularization_ is separate step, no edit of config file is required for that. You pass your photos, you get your depth maps

Prove me wrong, I'll appreciate that

**Comment by Vilour** [comment]
> > with the regularization of Depth-Anything
> 
> [Depth Anything](https://depth-anything.github.io/) is not good enough compared to [Depth Anything V2](https://depth-anything-v2.github.io/) nowadays
> 
> _Depth Regularization_ is separate step, no edit of config file is required for that. You pass your photos, you get your depth maps
> 
> Prove me wrong, I'll appreciate that

Thank you for your reply. I'm referring the config of CityGaussian-V2:


```
model:
  initialize_from: outputs/citygsv2_upper_coarse_sh2/checkpoints/epoch=48-step=30000.ckpt
  overwrite_config: False
  gaussian:
    class_path: internal.models.gaussian_2d.Gaussian2D
    init_args:
      sh_degree: 2
      optimization:
        means_lr_init: 0.000064
        means_lr_scheduler:
          lr_final: 0.00000064
        scales_lr: 0.004
  metric:
    class_path: internal.metrics.citygsv2_metrics.CityGSV2Metrics
    init_args:
      lambda_normal: 0.0125
      normal_regularization_from_iter: 0
      depth_loss_type: l1+ssim
      depth_loss_ssim_weight: 1.0
      depth_loss_weight:
        init: 0.5
        final_factor: 0.05
  renderer: 
    class_path: internal.renderers.sep_depth_trim_2dgs_renderer.SepDepthTrim2DGSRenderer
    init_args:
      depth_ratio: 1.0
      prune_ratio: 0.05
  density: internal.density_controllers.citygsv2_density_controller.CityGSV2DensityController
trainer:
  check_val_every_n_epoch: 20
  max_steps: 30000
data:
  path: data/GauU_Scene/CUHK_UPPER_COLMAP
  parser:
    class_path: EstimatedDepthBlockColmap
    init_args:
      down_sample_factor: 3.4175
      content_threshold: 0.05
      block_dim:  # removed z dimension
      - 3
      - 3
```

We have depth_loss from Depth-Anything-v2. Can we apple the depth_loss to 3DGS, instead of 2DGS?

**Comment by DekuLiuTesla** [comment]
@Vilour Sure, just by changing `model.gaussian.class_path`. You can refer to `configs/citygs_lfls_sh2_trim.yaml` to see the practice.

**Comment by Vilour** [comment]
> [@Vilour](https://github.com/Vilour) Sure, just by changing `model.gaussian.class_path`. You can refer to `configs/citygs_lfls_sh2_trim.yaml` to see the practice.
Thank you. I noticed the normal loss seems not available for 3DGS. How to modify the `render` and `metrics` accordingly?

**Comment by DekuLiuTesla** [comment]
@Vilour you can try `vanilla_trim_renderer` and maintain the metrics unchanged.
